data_handler.py

```diff
@@ -135,23 +137,23 @@ def _process_smiley_json(self) -> None:
 
                 # then ensure it hasn't already been processed prior to a crash, and
                 # that it should be processed at all cf. previously processed restaurants
-                if not temp_file.contains(restaurant['pnr']) \
-                        and prev_processed.should_process_restaurant(restaurant):
+                if prev_processed.should_process_restaurant(restaurant):
 
                     # only sleep if --no-scrape is not passed, and if our cvr provider requests it.
                     if not self._skip_scrape and self.cvr_handler.SHOULD_SLEEP and row_index > 0:
                         time.sleep(self.cvr_handler.CRAWL_DELAY)
 
                     # only collect data if we haven't passed --no-scrape
-                    processed = restaurant if self._skip_scrape \
-                        else self.cvr_handler.collect_data(restaurant)
+                    if not self._skip_scrape:
+                        restaurant = self.cvr_handler.collect_data(restaurant)
```

Comment:
The logic here will be bit different when #18 gets merged. See
https://github.com/sw814f21/filter_xml/blob/43a55f351a268be8c52f3592284f400109d29478/data_handler.py#L147-L151